src/opers.h

```diff
@@ -242,98 +261,82 @@ static inline Obj NEW_FLAGS(UInt len)
+*F  SET_HASH_FLAGS( <flags>, <hash> )  . . . . . .  set  hash value of <flags>
 */
-static inline UInt LEN_FLAGS(Obj flags)
+static inline void SET_HASH_FLAGS(Obj flags, UInt hash)
 {
-    return (SIZE_OBJ(flags) / sizeof(Obj) - 3) << LBIPEB;
-};
+    ((UInt *)ADDR_OBJ(flags))[2] = hash;
+}
 
 /****************************************************************************
 **
 *F  AND_CACHE_FLAGS( <flags> )  . . . . . . . . . 'and' cache of a flags list
 */
-#define AND_CACHE_FLAGS(list)           (CONST_ADDR_OBJ(list)[2])
+static inline Obj AND_CACHE_FLAGS(Obj flags) {
+    return CONST_ADDR_OBJ(flags)[0];
+}
 
 
 /****************************************************************************
 **
 *F  SET_AND_CACHE_FLAGS( <flags>, <len> ) set the 'and' cache of a flags list
 */
-#define SET_AND_CACHE_FLAGS(flags,andc)  (ADDR_OBJ(flags)[2]=(andc))
-
-
-/****************************************************************************
-**
-*F  NRB_FLAGS( <flags> )  . . . . . .  number of basic blocks of a flags list
-*/
-static inline UInt NRB_FLAGS(Obj flags)
-{
-    return SIZE_OBJ(flags) / sizeof(Obj) - 3;
-};
-
+static inline void SET_AND_CACHE_FLAGS(Obj flags, Obj andc) {
+    ADDR_OBJ(flags)[0]=andc;
+}
 
-/****************************************************************************
-**
-*F  BLOCKS_FLAGS( <flags> ) . . . . . . . . . . . . data area of a flags list
-*/
-#define BLOCKS_FLAGS(flags)             ((UInt*)(ADDR_OBJ(flags)+3))
 
 
 /****************************************************************************
 **
-*F  BLOCK_ELM_FLAGS( <list>, <pos> )  . . . . . . . .  block  of a flags list
-**
-**  'BLOCK_ELM_FLAGS' return the block containing the <pos>-th element of the
-**  flags list <list> as a UInt value, which is also a  valid left hand side.
-**  <pos>  must be a positive  integer  less than or  equal  to the length of
-**  <list>.
-**
-**  Note that 'BLOCK_ELM_FLAGS' is a macro, so do not call it  with arguments
-**  that have side effects.
+*F  TRUE_FLAGS( <flags>, <ix> )  the <ix>th set position in flags
+**                               caller is responsible for <ix> being in range
 */
-#define BLOCK_ELM_FLAGS(list, pos)      (BLOCKS_FLAGS(list)[((pos)-1) >> LBIPEB])
 
+static inline UInt TRUE_FLAGS(Obj flags, UInt ix) {
+    return (UInt) ((const UInt2 *)(CONST_ADDR_OBJ(flags) + 3))[ix];
+}
 
 /****************************************************************************
 **
-*F  MASK_POS_FLAGS( <pos> ) . . .  . .  bit mask for position of a flags list
-**
-**  'MASK_POS_FLAGS(<pos>)' returns a UInt with a single set  bit in position
-**  '(<pos>-1) % BIPEB',
-**  useful for accessing the <pos>-th element of a 'FLAGS' list.
-**
-**  Note that 'MASK_POS_FLAGS'  is a macro, so  do not call it with arguments
-**  that have side effects.
+*F  SET_TRUE_FLAGS( <flags>, <ix>, <filt> )  the <ix>th set position in flags
+**                               caller is responsible for <ix> being in range
```

Comment:
What does that mean? I.e. what is "in range" in the new model? I assume it means that it doesn't leave the bag -- OK, so let's add a `GAP_ASSERT` to check that anyway?